src/go/rpk/pkg/cli/cmd/topic/config.go

```diff
@@ -10,65 +10,112 @@
 package topic
 
 import (
-	"errors"
-	"fmt"
-	"strings"
+	"context"
 
-	"github.com/Shopify/sarama"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
 )
 
-func NewSetConfigCommand(
-	admin func() (sarama.ClusterAdmin, error),
-) *cobra.Command {
+func NewAlterConfigCommand(fs afero.Fs) *cobra.Command {
+	var (
+		sets []string // key=val
+		dels []string // key only
+		adds []string // key=val
+		rems []string // key=val
+
+		dry bool
+	)
+
 	cmd := &cobra.Command{
-		Use:   "set-config <topic> <key>=<value> [<key>=<value>...]",
-		Short: "Set the topic's config key/value pairs",
-		Args: func(_ *cobra.Command, args []string) error {
-			if len(args) < 2 {
-				return errors.New("a topic name and at least one key=value pair is required")
+		Use:   "alter-config [TOPICS...] --set key=value --del key2,key3",
+		Short: "Set, delete, add, and remove key/value configs for a topic.",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := kafka.NewFranzClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer cl.Close()
+
+			if len(topics) == 0 {
+				out.Exit("No topics were requested to incrementally change the configuration of!")
 			}
-			return nil
-		},
-		// We don't want Cobra printing CLI usage help if the error isn't about CLI usage.
-		SilenceUsage: true,
-		RunE: func(cmd *cobra.Command, args []string) error {
 
-			topicName := args[0]
+			setKVs, err := parseKVs(sets)
+			out.MaybeDie(err, "unable to parse --set: %v", err)
+			addKVs, err := parseKVs(adds)
+			out.MaybeDie(err, "unable to parse --add: %v", err)
+			remKVs, err := parseKVs(rems)
```

Comment:
Do we need to `parseKVs` here since they values aren't supposed to be `k=v`?

Reply:
There are removals, i.e. removing something from a list. Kafka itself uses the terms "append" and "subtract", but I'm not sure subtract is clearer (and I think add is just as clear as append).

I can rename the variables so that rems & dels are not so close visually (change this to removals & deletes).

Reply:
Ahhh, I see. Up to you, maybe adding a comment is enough.